core/src/mender-client.c

```diff
@@ -1383,30 +1356,21 @@ mender_client_download_artifact_callback(char *type, cJSON *meta_data, char *fil
     }
 
     /* Retrieve ID and artifact name */
-    cJSON *json_id = NULL;
-    if (NULL == (json_id = cJSON_GetObjectItemCaseSensitive(mender_client_deployment_data, "id"))) {
-        mender_log_error("Unable to get ID from the deployment data");
-        goto END;
-    }
-    char *id;
-    if (NULL == (id = cJSON_GetStringValue(json_id))) {
+    const char *id;
+    if (MENDER_OK != mender_deployment_data_get_id(mender_client_deployment_data, &id)) {
         mender_log_error("Unable to get ID from the deployment data");
         goto END;
     }
-    cJSON *json_artifact_name = NULL;
-    if (NULL == (json_artifact_name = cJSON_GetObjectItemCaseSensitive(mender_client_deployment_data, "artifact_name"))) {
-        mender_log_error("Unable to get artifact name from the deployment data");
-        goto END;
-    }
-    char *artifact_name;
-    if (NULL == (artifact_name = cJSON_GetStringValue(json_artifact_name))) {
+    const char *artifact_name;
+    if (MENDER_OK != mender_deployment_data_get_artifact_name(mender_client_deployment_data, &artifact_name)) {
         mender_log_error("Unable to get artifact name from the deployment data");
         goto END;
     }
 
     /* Invoke update module download callback */
-    struct mender_update_download_state_data_s download_state_data = { id, artifact_name, type, meta_data, filename, size, data, index, length, false };
-    mender_update_state_data_t                 state_data          = { .download_state_data = &download_state_data };
+    struct mender_update_download_state_data_s download_state_data
+        = { (char *)id, (char *)artifact_name, type, meta_data, filename, size, data, index, length, false };
```

Comment:
@vpodzime should these be `const char *` in the `mender_update_download_state_data_s` struct?

Reply:
Yes, we don't want the struct to modify or own those strings. :+1:

Reply:
Are you changing this?

Reply:
Will do ;)